R/scripts.R

```diff
@@ -99,12 +106,31 @@ run_civis <- function(expr, ...) {
 #'
 #' # Run the template
 #' run_template(id, arguments = list(arg1 = 1, arg2 = 2), ...)
+#'
+#' # Run the template and return JSON value outputs
+#' run_template(id, arguments = list(arg1 = 1, arg2 = 2), JSONValue=TRUE...)
 #' }
-run_template <- function(id, arguments, ...) {
+run_template <- function(id, arguments, JSONValue=FALSE, ...) {
   job <- scripts_post_custom(id, arguments = arguments, ...)
   run <- scripts_post_custom_runs(job$id)
   await(scripts_get_custom_runs, id = job$id, run_id = run$id)
-  fetch_output_file_ids(civis_script(job$id, run$id))
+  if (JSONValue) {
+     output <- fetch_output(civis_script(job$id, run$id))
+     json_output <- output[sapply(output, function(o) o$objectType=="JSONValue")]
+     if (length(json_output) > 1) {
+        cat("Error in returning JSON outputs of template run -- too many JSON outputs")
+	return()
+     }
+     if (length(json_output) == 0) {
+        cat("Error in returning JSON outputs of template run -- no JSON output")
+	return()
+     }
+     return(json_output[[1]]$value)
```

Comment:
Returning something if and only if there is exactly 1 JSON output

Reply:
My instinct here (if `JSONValue=`TRUE`) would be the following:

* If there is 1 JSONValue return that value
* If there is more than 1 JSONValue, return the first and warn that only the first is returned.
* If there are no JSONValues returned, warn that no value was returned and return `list()`

Reply:
That makes sense as well. I wasn't positive what the best behavior was for case 2. 
In the latest commit, I changed it to the behavior described.